pkg/mover/intermediate.go

```diff
@@ -119,9 +129,113 @@ func tarImages(imageChanges []*internal.ImageChange, logger Logger) (string, err
 		logger.Printf("Processing image %s\n", change.ImageReference.Name())
 	}
 
-	logger.Printf("Writing all %d images...\n", len(refToImage))
+	logger.Printf("Writing %d images...\n", len(refToImage))
 	if err := tarball.MultiRefWrite(refToImage, imagesFile); err != nil {
 		return "", err
 	}
 	return imagesFile.Name(), nil
 }
+
+// IsIntermediateBundle returns tue only if VerifyIntermediateBundle finds no errors
+func IsIntermediateBundle(bundlePath string) bool {
+	return VerifyIntermediateBundle(bundlePath) == nil
+}
+
+type fileValidations struct {
+	filename, format string
+	validate         func(io.Reader) error
+}
+
+// VerifyIntermediateBundle returns true if the path points to an uncompressed
+// tarball with:
+//  A hints.yaml YAML file
+//  A manifest.json for the images
+//  A directory container an unpacked chart directory with valid YAMLs Chart.yaml & values.yaml
+func VerifyIntermediateBundle(bundlePath string) error {
+	validations := []fileValidations{
+		{filename: "hints.yaml", format: "YAML", validate: validateYAML},
+		{filename: originalChart + "/Chart.yaml", format: "YAML", validate: validateYAML},
+		{filename: originalChart + "/values.yaml", format: "YAML", validate: validateYAML},
+		{filename: imagesTar, format: "TAR", validate: validateTAR},
```

Comment:
how expensive it this validation?

Reply:
The Tar validation is probably the cheapest, we only validate we can read the first header properly. That means the file must be a valid tar, at least initially. It might be broken later, but we would detect that anyway.

The YAML validation is more expensive as it reads the whole file. But those files are usually not too big. Hints we load in memory completely.

I understand we might not want to make such a complete validation at this point. But then file naming would be enough. We already fail if any of the expected files is missing.

I can simplify if needed.

Reply:
please do

Reply:
done